### PR TITLE
Add generated publish API.

### PR DESCRIPTION
**Pull request by OleksiiZubko**
Adding the generated from OpenAPI spec. functions,
interfaces and models.

Adding unit tests for the added functions.
The test checks the correct building URL and request type for the request
by the incoming params.

Resolves: OLPEDGE-2005

Signed-off-by: Oleksii Zubko <ext-oleksii.zubko@here.com>

### CODE REVIEW

**Comment by mykhailo-kuchma** [comment]
All parameters are optional?

**Comment by OleksiiZubko** [comment]
> All parameters are optional?

not all, but some interfaces have optional props. as other generated apis

**Comment by OleksiiZubko** [comment]
> > All parameters are optional?
> 
> not all, but some interfaces have optional props. as other generated apis

2checked and updated. Some should be optional, some made required

File: @here/olp-sdk-dataservice-api/lib/publish-api-v2.ts [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+    expires: number;
+}
+
+export type StateEnum =
+    | "initialized"
+    | "submitted"
+    | "cancelled"
+    | "failed"
+    | "succeeded"
+    | "expired";
```
[the remark this answers is not in the document]
Comment (OleksiiZubko) [reply @here/olp-sdk-dataservice-api/lib/publish-api-v2.ts:82]: It's generated from spec. We have only that in spec. Are you proposing just use "string" as a param?

[the remark this answers is not in the document]
Comment (andescu) [reply @here/olp-sdk-dataservice-api/lib/publish-api-v2.ts:82]: Yes, I would propose to just use string.

[the remark this answers is not in the document]
Comment (OleksiiZubko) [reply @here/olp-sdk-dataservice-api/lib/publish-api-v2.ts]: updated, tnx.

**Pull request merged**